solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/LTRScoringAlgorithm.java

```diff
@@ -169,5 +198,24 @@ public abstract Explanation explain(LeafReaderContext context, int doc,
   public String toString() {
     return  getClass().getSimpleName() + "(name="+getName()+")";
   }
+  
+  /**
+   * Goes through all the stored feature values, and calculates the normalized
+   * values for all the features that will be used for scoring.
+   */
+  private void normalizeFeatures(float[] modelFeatureValues) {
+    modelFeatureValuesNormalized = modelFeatureValues.clone();
+    for(int idx = 0; idx < modelFeatureValuesNormalized.length; ++idx) {
+      if (idx < norms.size()) {
```

Comment:
this should never happen right?  do we need this check in the for loop? Can we check the sizes once outside and throw an exception since it should never ever happen?